pkg/cnispawn/netns.go

```diff
@@ -19,25 +19,48 @@ package cnispawn
 import (
 	"fmt"
 	"io/ioutil"
+	"log"
 	"os"
 	"os/exec"
 	"path"
 	"strings"
 
 	"github.com/containernetworking/plugins/pkg/ns"
 	"github.com/kinvolk/kube-spawn/pkg/bootstrap"
+	"github.com/kinvolk/kube-spawn/pkg/utils"
 )
 
 var (
-	gopath  string = os.Getenv("GOPATH")
-	cniPath string = os.Getenv("CNI_PATH")
+	homePath string = os.Getenv("HOME")
+	goPath   string = os.Getenv("GOPATH")
+	cniPath  string = os.Getenv("CNI_PATH")
 )
 
 type CniNetns struct {
 	netns ns.NetNS
 }
 
 func NewCniNetns() (*CniNetns, error) {
+	if !utils.IsValidDir(goPath) {
+		// fall back to $HOME/go
+		goPathOrig := goPath
+		goPath = path.Join(homePath, "go")
+		log.Printf("cnispawn: invalid GOPATH %q, fall back to %s...\n", goPathOrig, goPath)
+		if !utils.IsValidDir(goPath) {
+			return nil, fmt.Errorf("cnispawn: invalid GOPATH %q", goPath)
+		}
+	}
+
+	if !utils.IsValidDir(cniPath) {
+		// fall back to $GOPATH/bin
+		cniPathOrig := cniPath
+		cniPath = path.Join(goPath, "bin")
+		log.Printf("cnispawn: invalid CNI_PATH %q, fall back to %s...\n", cniPathOrig, cniPath)
+		if !utils.IsValidDir(cniPath) {
+			return nil, fmt.Errorf("cnispawn: invalid CNI_PATH %q", cniPath)
+		}
+	}
+
```

Comment:
not sure but could all of this go into `pkg/util` so you have something like this?
```
goPath, err := util.ValidateGoPath(os.Getenv("GOPATH"))
if err ...
cniPath, err := util.ValidateCniPath(...)
```

Because it looks like we have things double...